src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -2069,98 +2072,129 @@ EquivalenceKind TypeObjectUtils::get_map_component_equiv_kind_for_consistency(
     }
     else if (is_indirect_hash_type_identifier(identifier))
     {
-        if (is_key)
-        {
-            // indirecr hash for key is not allowed
-            return TK_NONE;
-        }
         EquivalenceKind element_equiv_kind;
         EquivalenceKind key_equiv_kind;
 
         switch (identifier._d())
         {
             case TI_PLAIN_SEQUENCE_SMALL:
-                return get_map_component_equiv_kind_for_consistency(*identifier.seq_sdefn().element_identifier(),
-                               false);
+                if ((*identifier.seq_sdefn().element_identifier())._d() == TK_NONE)
+                {
+                    return TK_NONE;
+                }
```

Comment:
This case is the first thing checked in this recursive method. Why not call directly the method removing this check? Please, apply to the other similar implementations.